p2p/node/streamManager/streamManager.go

```diff
@@ -147,23 +150,30 @@ func (sm *basicStreamManager) OpenStream(peerID p2p.PeerID) error {
 	if _, ok := sm.streamCache.Get(peerID); ok {
 		return nil
 	}
-	// Create a new stream to the peer and register it in the cache
-	stream, err := sm.host.NewStream(sm.ctx, peerID, quaiprotocol.ProtocolVersion)
-	if err != nil {
-		return fmt.Errorf("error opening new stream with peer %s", peerID)
-	}
-	wrappedStream := streamWrapper{
-		stream:    stream,
-		semaphore: make(chan struct{}, c_maxPendingRequests),
-		errCount:  0,
-	}
-	sm.streamCache.Add(peerID, wrappedStream)
-	go quaiprotocol.QuaiProtocolHandler(stream, sm.p2pBackend)
-	log.Global.WithField("PeerID", peerID).Info("Had to create new stream")
-	if streamMetrics != nil {
-		streamMetrics.WithLabelValues("NumStreams").Inc()
+	timer := time.NewTimer(c_stream_timeout)
+	defer timer.Stop()
+	select {
+	case <-timer.C:
+		return errors.New("stream creation timeout")
```

Comment:
I don't think this does anything, you can't kill a process while it's running in this way